### ISSUE TITLE
How does it decide which segment is valid one?

### ISSUE THREAD

**Issue by AnatolyKap** [luqmanahmad/ignite-plugins#1]
Hi

I have a theoretical question.
As I can see your implementation of SegmentationResolver.isValidSegment() simply checks access to a shared resource (i.p. shared file folder). 
But what if after segmentation nodes from both network segments still have access to the folder? 
Is not it necessary to define somehow which segment is valid and which is not?

Thanks

**Comment by luqmanahmad** [comment]
AnatolyKap, 

In case you are using [SharedFileSystemSegmentationResolver](https://github.com/luqmanahmad/ignite-plugins/blob/master/network-segmentation/src/main/java/com/ig/segmentation/network/segment/SharedFileSystemSegmentationResolver.java) it checks for a shared resource on the network. 

Considering your topology consists of 3 nodes (A,B,C) and segmentation happens on Node A for any reason then Node A will behave according to [SegmentationPolicy](https://ignite.apache.org/releases/latest/javadoc/org/apache/ignite/plugin/segmentation/SegmentationPolicy.html) you have defined. **Default SegmentationPolicy** is [STOP](https://ignite.apache.org/releases/latest/javadoc/org/apache/ignite/configuration/IgniteConfiguration.html#DFLT_SEG_PLC) which means it will stop the Node A and the new topology will contain only Node B and C.

In short, when the segment is not valid the node behaves according to SegmentationPolicy. I will suggest you to read java docs of SegmentationPolicy.

Makes sense?

**Comment by AnatolyKap** [comment]
Problem is the "shared resource on the network" would be single point of failure for whole ignite cluster
I use AWS and struggling to find a shared resource which I could rely on

**Comment by luqmanahmad** [comment]
Have you considered using [TcpIpSegmentationResolver](https://github.com/luqmanahmad/ignite-plugins/blob/master/network-segmentation/src/main/java/com/ig/segmentation/network/segment/TcpIpSegmentationResolver.java)?

You can use TcpIpSegmentationResolver and do a simple ping to your server (Database or applications etc) - It uses java sockets behind the scenes to connect to a specific resource and closes the connection immediately after connecting to it. See this [example 1](https://www.programcreek.com/java-api-examples/?class=java.net.Socket&method=connect) on this page to understand more.

**Comment by AnatolyKap** [comment]
> your server (Database or applications etc) 

this server would be single point of failure

**Comment by luqmanahmad** [comment]
Correct, so in AWS for example you can point to your RDS instance and if any node is unable to ping it the segment will not be validated. Let me know if it makes sense to you?

**Comment by AnatolyKap** [comment]
We don't use RDS. We use active-active pair of Oracle databases with Oracle Golden Gate two-ways replication.

**Comment by luqmanahmad** [comment]
As long there is a single point of failure which can be accessed through each node it should be ok to use TcpIpSegmentationResolver. I personally think, this will be the best choice for you.
